include/dali/c_api.h

```diff
@@ -16,221 +16,268 @@
 #define DALI_C_API_H_
 
 #include <cuda_runtime_api.h>
-#include <inttypes.h>
+#include <cinttypes>
```

Comment:
```suggestion
#include <inttypes.h>
```
This is supposed to be C API!

Reply:
Add a file `c_api_build_test.c` which would include c_api.h just to see if it compiles in C.

Reply:
I wanted to do that, but we decided to hold our horses on reworking this to be actually a C api, not a "C-like" api.

Reply:
"Supposed" ;)
I agree and will do it but in a followup PR: right now there are `bool`s used in this API, so let's make it into the next one

This error fixed.